docarray/index/backends/hnswlib.py

```diff
@@ -429,30 +429,13 @@ def num_docs(self) -> int:
             self._num_docs = self._get_num_docs_hnsw()
         return self._num_docs
 
-    # def _get_num_docs_hnsw(self) -> int:
-    #     """
-    #     Get the number of documents using the HNSW method.
-    #     """
-    #     first_index_key = next(iter(self._hnsw_indices))  # Get the first key
-    #     first_hnsw_index = self._hnsw_indices[first_index_key]
-    #     hnsw_num_docs = first_hnsw_index.element_count
-    #     return hnsw_num_docs
-
-    def _count_docs_in_hnswlib(self) -> int:
-        """
-        Get the number of documents using the HNSW method from the first index in the dictionary.
+    def is_index_empty(self) -> bool:
         """
-        # Get the first index from the dictionary
-        first_index_key = next(iter(self._hnsw_indices))
-        first_hnsw_index = self._hnsw_indices[first_index_key]
-
-        # Use HNSWlib's method to count documents
-        hnsw_num_docs = first_hnsw_index.element_count
-        return hnsw_num_docs
+        Check if the index is empty by comparing the number of documents to zero.
 
-    def is_index_empty(self) -> bool:
-        # Check if the index is empty by counting documents in HNSWlib
-        return self._count_docs_in_hnswlib() == 0
+        :return: True if the index is empty, False otherwise.
```

Comment:
this should be implemented on the base Document Index class. Not here for now

Reply:
Should I do this like this : 
class BaseDocIndex:
    # ... other methods and attributes ...

    def is_index_empty(self) -> bool:
        """
        Check if the index is empty by comparing the number of documents to zero.

        :return: True if the index is empty, False otherwise.
        """
        return self.num_docs() == 0

# ... other code ...

class HNSWLibIndex(BaseDocIndex):
    # ... existing code ...

    def _count_docs_in_hnswlib(self) -> int:
        # Use HNSWlib's method to count documents
        return self._hnsw_indices.element_count

    def is_index_empty(self) -> bool:
        # Check if the index is empty by counting documents in HNSWlib
        return self._count_docs_in_hnswlib() == 0

    # Modify other method calls that previously used num_docs to use _count_docs_in_hnswlib

    # ... other methods ...

Reply:
For now, just let's do the first part of the code, and then we go after the other.

Reply:
okay, then I should just do this : 
class BaseDocIndex:
    # ... other methods and attributes ...

    def is_index_empty(self) -> bool:
        """
        Check if the index is empty by comparing the number of documents to zero.

        :return: True if the index is empty, False otherwise.
        """
        return self.num_docs() == 0

is this approach right?

Reply:
yes, let's do this. And use this method in the parts where `self.num_docs() == 0` is used internally in the indexes